client/rest/src/main/java/org/elasticsearch/client/ResponseListener.java

```diff
@@ -23,6 +23,10 @@
  * Listener to be provided when calling async performRequest methods provided by {@link RestClient}.
  * Those methods that do accept a listener will return immediately, execute asynchronously, and notify
  * the listener whenever the request yielded a response, or failed with an exception.
+ *
+ * <p>
+ * Note that it is <strong>not</strong> safe to call {@link RestClient#close()} from either of these
+ * callbacks.
```

Comment:
I never thought somebody would do this. But I see how it happened with reindex, and it may definitely happen again...